### ISSUE TITLE
"Safely use '===' here" is bad advice for string comparisons involving objects with __toString() magic methods

### ISSUE THREAD

**Issue by LordPachelbel** [kalessil/phpinspectionsea#1844]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), 4.0.7.1 |
| Language level | PHP 8.1 (but it happens in older versions, too)                    |

Current behaviour
--
Reporting "Safely use '===' here" is bad advice when the comparison involves objects with a `__toString()` magic method, such as SimpleXMLElement nodes, because `__toString()` isn't implicitly called during `===` and `!==` comparisons, only during the weaker `==` and `!=` comparisons. To use `===` and `!==` you have to explicitly cast the text as a string first.

Example using SimpleXML
--

Example XML data:
```XML
<navigation>
  <page id="node0" p_id="node0">
    <uri>index.html</uri>
    <name>Home</name>
    <hidden>No</hidden>
    <page id="node1" p_id="node1">
      <uri>subpage1.html</uri>
      <name>Subpage 1</name>
      <hidden>No</hidden>
    </page>
    <page id="node2" p_id="node2">
      <uri>subpage2.html</uri>
      <name>Subpage 2</name>
      <hidden>Yes</hidden>
    </page>
  </page>
</navigation>
```

Example usage:
```PHP
$xml = simplexml_load_string(file_get_contents('navigation.xml'));
foreach($xml -> page -> page as $subpage) {
  if($subpage->hidden == 'Yes') { // `__toString()` is implicitly called here
    // do stuff
  }
}
```

The inspector will report `Safely use '===' here.` for that if-statement, but if you use `===` it will always be false, because `$subpage->hidden` isn't a string, it's a SimpleXMLElement object.

```PHP
  if($subpage->hidden === 'Yes') { // `__toString()` is NOT implicitly called here, so this will always be false
    // do stuff
  }
```

To do a `===` or `!==` comparison you have to cast it as a string first:
```PHP
  if((string) $subpage->hidden === 'Yes') {
    // do stuff
  }
```

If it's possible, it would be nice if the inspector would provide a different recommendation when either side of `==` and `!=` comparisons are objects with a `__toString()` method. Maybe a slightly expanded version of the existing message? E.g.
* "Safely use (string) ==="
* "Safely use === (string)"
* "Safely use (string) === (string) here"
* etc.
